pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -139,7 +140,8 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self.setup_precision_plugin()
         optimizers_to_device(self.optimizers, self.root_device)
 
-        if self.sync_batchnorm:
+        trainer_fn = self.lightning_module.trainer.state.fn
+        if trainer_fn == TrainerFn.FITTING and self.sync_batchnorm:
```

Comment:
setup has a `trainer` input
```suggestion
        if trainer.state.fn == TrainerFn.FITTING and self.sync_batchnorm:
```